File: ecom-back/src/main/java/com/ecom/models/Product.java
package com.ecom.models;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name="ecom_products")
public class Product {

	
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int productId;
	
	@Column(name = "product_brand_name",length = 300,unique = true)
	private String productName;
	
	private String productDesc;
	
	private double productPrice;
	
	private int productQuantity;
	
	private boolean live;
	
	private boolean stock=true;
	
	private String imageName;
	
	@ManyToOne(fetch = FetchType.EAGER)
	private Category category;
	
	
	public Product() {
		super();
	}

	public int getProductId() {
		return productId;
	}

	public void setProductId(int productId) {
		this.productId = productId;
	}

	public String getProductName() {
		return productName;
	}

	public void setProductName(String productName) {
		this.productName = productName;
	}

	public String getProductDesc() {
		return productDesc;
	}

	public void setProductDesc(String productDesc) {
		this.productDesc = productDesc;
	}

	public double getProductPrice() {
		return productPrice;
	}

	public void setProductPrice(double productPrice) {
		this.productPrice = productPrice;
	}

	public boolean isStock() {
		return stock;
	}

	public void setStock(boolean stock) {
		this.stock = stock;
	}

	public boolean isLive() {
		return live;
	}

	public void setLive(boolean isLive) {
		this.live = isLive;
	}

	public String getImageName() {
		return imageName;
	}

	public void setImageName(String imageName) {
		this.imageName = imageName;
	}

	public int getProductQuantity() {
		return productQuantity;
	}

	public void setProductQuantity(int productQuantity) {
		this.productQuantity = productQuantity;
	}

	public Category getCategory() {
		return category;
	}

	public void setCategory(Category category) {
		this.category = category;
	}
	
	
	
	
	
	
	
	
	
}

File: angular-ecom/src/app/models/Product.ts
export class Product {
    productName: string;
    productDesc: string;
    productPrice: number;
    productQuantity: number;
    live: boolean;
    stock: boolean;
    imageName: string;
    categoryId: number;


    constructor(productName: string, productDesc: string, productPrice: number, productQuantity: number, live: boolean, stock: boolean, imageName: string, categoryId: number) {
        this.productName = productName
        this.productDesc = productDesc
        this.productPrice = productPrice
        this.productQuantity = productQuantity
        this.live = live
        this.stock = stock
        this.imageName = imageName
        this.categoryId = categoryId
    }





    public getProductName(): string {
        return this.productName;
    }

    public getProductDesc(): string {
        return this.productDesc;
    }

    public getProductPrice(): number {
        return this.productPrice;
    }

    public getProductQuantity(): number {
        return this.productQuantity;
    }

    public getLive(): boolean {
        return this.live;
    }

    public getStock(): boolean {
        return this.stock;
    }

    public getImageName(): string {
        return this.imageName;
    }

    public getCategoryId(): number {
        return this.categoryId;
    }




}